### PR TITLE
feat: additional options when assuming roles

### PR DESCRIPTION
**Pull request by sumupitchayan**
Required to implement [session tags](https://github.com/aws/aws-cdk/issues/26157) and a prerequisite to https://github.com/aws/aws-cdk/pull/31089.

### CODE REVIEW

**Comment by iliapolo** [comment]
@TheRealAmazonKendra Do you need to mark this as a breaking change so it will do a major version bump? or does this happen automatically?

**Comment by TheRealAmazonKendra** [comment]
> @TheRealAmazonKendra Do you need to mark this as a breaking change so it will do a major version bump? or does this happen automatically?

It should happen automatically whenever the schema changes.

File: lib/manifest.ts [hunk 1]
```diff
@@ -250,6 +247,30 @@ export class Manifest {
...
+    _key: string,
+    _schema: jsonschema.Schema,
+    _options: jsonschema.Options,
+    _ctx: jsonschema.SchemaContext
+  ) {
+    if (_key !== 'assumeRoleAdditionalOptions') {
+      return;
+    }
+
+    const assumeRoleOptions = instance.assumeRoleAdditionalOptions;
```
Comment (iliapolo) [inline lib/manifest.ts:265]: Been going back and forth on whether this belongs in the CLI or here. 

- **Argument for CLI**: The CLI is the deployment tool that actually suffers from this duplication, why are we preventing other deployment tools from creating such manifests?
- **Argument for here**: Creating a manifest with both `assumeRoleArn` and `assumeRoleAdditionalOptions.RoleArn` is confusing, and we should disallow it altogether, regardless of which deployment tool will act on the schema. 

I think its unlikely other deployment tools will choose to ignore this duplication, so I put it here. It also has the benefit of not duplicating the validation on both the CLI and cdk-assets. 

Interested in thoughts here though.

Comment (rix0rrr) [reply lib/manifest.ts:265]: I think it makes sense to consider this part of the schema.

We could/would express this as part of the JSON Schema if we could, so treating it equivalently seems solid to me.

**Review by rix0rrr — APPROVED** [review]

**Review by rix0rrr — APPROVED** [review]

**Comment by iliapolo** [comment]
@Mergifyio refresh

**Pull request merged**